add container replicator relationship to item view of containers and pods.

moolitayer opened the pull request.

moolitayer commented:
[before] container:
![screenshot from 2015-08-04 17-33-54](https://cloud.githubusercontent.com/assets/3010449/9063260/43355ce4-3acf-11e5-87dd-0a0dc8b30aac.png)

[before] container group:
![screenshot from 2015-08-04 17-33-44](https://cloud.githubusercontent.com/assets/3010449/9063280/597d867a-3acf-11e5-90c2-6f37e54d5da5.png)

[after] container:
![screenshot from 2015-08-04 17-30-42](https://cloud.githubusercontent.com/assets/3010449/9063342/8dab7768-3acf-11e5-8c00-3cc01611421c.png)

[after] container group:
![screenshot from 2015-08-04 17-31-33](https://cloud.githubusercontent.com/assets/3010449/9063325/7c10934e-3acf-11e5-9430-7fd273985ff4.png)

moolitayer commented:
fyi @simon3z @abonas 
@miq-bot add_label ui, enhancement, providers/containers

abonas commented:
@matthewd can you perhaps provide an insight on the rubocop failures above? they are not part of the changed lines in this patch, and they seem to be added by a refactoring you've done lately. thank you.

abonas commented:
@moolitayer the title says "route" but the code says "replicator"

simon3z commented on app/models/container.rb:
```diff
@@ -4,6 +4,7 @@ class Container < ActiveRecord::Base
 
   has_one    :container_group, :through => :container_definition
   delegate   :ext_management_system, :to => :container_group
+  delegate   :container_replicator, :to => :container_group
```
Can you check if all these ones can be re-written with `:through =>`? Thanks /cc @abonas

blomquisg replied:
@simon3z, I think `:to` is the correct syntax here for `delegate`.

http://api.rubyonrails.org/classes/Module.html#method-i-delegate

simon3z replied:
@blomquisg yes I was questioning the use of `delegate` vs `:through`, e.g.:

    has_one :container_replicator, :through => :container_group

It should work right? (preferred too?)

matthewd commented:
@abonas I sure can!

The relevant insight is: ignore them :smile:

The bot's rubocop seems to currently be tripping over the newer ruby syntax used in this file.

blomquisg commented:
LGTM ... @abonas, @simon3z, good to go?

moolitayer commented:
@abonas fixed title mixup.
@simon3z @blomquisg 
has_one :container_replicator, :through => :container_group
works fine too.

simon3z commented:
> @simon3z @blomquisg 
> has_one :container_replicator, :through => :container_group
> works fine too.

@blomquisg your call on what to do (switch to `has_one :container_replicator, :through => :container_group`? personally I'd prefer it)

blomquisg commented:
@simon3z, @moolitayer ... ah, yeah ... I don't have a preference on `delegate` vs. `has_one through`.  We have precedence for both throughout the app.

@matthewd, do you have any opinion here?

simon3z commented:
@blomquisg @matthewd my reasoning here is that with `:through` activerecord may have more information for building an efficient query.

matthewd commented:
@simon3z :+1:

It's unlikely to matter very often, but using hmt at least means that it can be used in `includes`.

abonas commented:
what is hmt?

On Fri, Aug 7, 2015 at 5:58 AM, Matthew Draper <notifications@github.com>
wrote:

> @simon3z <https://github.com/simon3z> [image: :+1:]
>
> It's unlikely to matter very often, but using hmt at least means that it
> can be used in includes.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ManageIQ/manageiq/pull/3715#issuecomment-128573387>.
>

matthewd commented:
Sorry.. `has_many :through`

matthewd commented:
... which I now realise was entirely confusing, because this is actually a `has_one :through`... oops :blush:

Anyway, yeah, do that. :+1:

abonas commented:
yes, I thought at first that hmt might be has_many through, but since it
was indeed has_one here, I wasn't sure what you meant :)

On Fri, Aug 7, 2015 at 12:35 PM, Matthew Draper <notifications@github.com>
wrote:

> ... which I now realise was entirely confusing, because this is actually a has_one
> :through... oops [image: :blush:]
>
> Anyway, yeah, do that. [image: :+1:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ManageIQ/manageiq/pull/3715#issuecomment-128656252>.
>

abonas commented:
> ... which I now realise was entirely confusing, because this is actually a has_one :through... oops :blush:
> 
> Anyway, yeah, do that. :+1:

@moolitayer please follow this recommendation and change the code. thanks.

moolitayer commented:
addressed comment:
-  delegate   :container_replicator, :to => :container_group
+  has_one    :container_replicator, :through => :container_group
(tested again)

simon3z commented:
:+1:  ready for merge @chessbyte @blomquisg
